### ISSUE TITLE
FontAwesome 5 fonts are not Debian-compatible

### ISSUE THREAD

**Issue by juliangilbey** [spyder-ide/qtawesome#220]
Hi!

I hadn't been aware of this issue until very recently, but it turns out that when FontAwesome became commercial and released version 5 (a completely fresh, redesigned set of fonts, not forked from version 4.7), they made their build system proprietary and closed-source, so Debian will not package FontAwesome 5 fonts. Here are some discussions about this:

* The Debian bug report that alerted me to the issue: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1025000
* The initial Debian bug report about FontAwesome 5 with contributions from the FontAwesome developers: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=902981
* A short discussion on the FontAwesome GitHub issue tracker: https://github.com/FortAwesome/Font-Awesome/issues/13467
* The ForkAwesome package: https://github.com/ForkAwesome/Fork-Awesome

Either my colleagues or I will have a go at using the Fork-Awesome FontAwesome compatibility layer over the next week or so to see whether this will allow us to continue using Spyder smoothly with a DFSG-free version of qtawesome; I really hope it turns out to be successful!  If you would like to look into moving away from FontAwesome and to Fork-Awesome, that would be amazing for us all!

Thanks as ever for your work on this.

**Comment by dalthviz** [comment]
Also, other approach that could help with this (and I believe was discussed before too in order reduce the current package size) is to reduce the bundled fonts and instead create a way to load fonts via entry-points. So with that, instead of bundling here a lot of fonts, each font should be in its own package and then if you want to use it you would need to explicitly install it along side QtAwesome.

What do you think about this other option for a future v2.0.0 @ccordoba12 ?

**Comment by juliangilbey** [comment]
Thanks @dalthviz! That is really useful to know.
I'm also looking into the Material Design Icons, though that's a separate matter: I don't currently understand why there are two different MDI icon fonts in this package, as they only release one icon font.  But I'll take a further look over the next week or two...

**Comment by dalthviz** [comment]
> I don't currently understand why there are two different MDI icon fonts in this package

That's because `mdi` is for version 5.x and `mdi6` is for version 6.x (similar to the management we have for FontAwesome: `fa` is version 4.x and `fa5*` is for version 5.x).

If you need any help or further clarification on the package and fonts available let us know!

**Comment by ccordoba12** [comment]
> Checking Fork-Awesome, seems like it generates a single font file and maybe has less icons than FontAwesome 5, right?
> 
> What do you think @ccordoba12 ?

This would introduce a regression for our users, so I'm -1 on it. As fas as I know, we're not violating FontAwesome license, so this is a Debian only problem which they need to address on their own.

> What do you think about this other option for a future v2.0.0 @ccordoba12 ?

That's an interesting idea but I don't know if it'd work without being online. So, I'm not sure about it.

**Comment by juliangilbey** [comment]
> > Checking Fork-Awesome, seems like it generates a single font file and maybe has less icons than FontAwesome 5, right?
> 
> This would introduce a regression for our users, so I'm -1 on it. As fas as I know, we're not violating FontAwesome license, so this is a Debian only problem which they need to address on their own.

Fair enough; I just wanted to alert you to it.  Knowing that Spyder doesn't actually use FontAwesome is a real help here - there are only a couple of other packages in Debian currently using qtawesome, so we are in good shape.

> > What do you think about this other option for a future v2.0.0 @ccordoba12 ?
> 
> That's an interesting idea but I don't know if it'd work without being online. So, I'm not sure about it.

I'm not sure how being online would make a difference, as the font is being served locally.

> >   I don't currently understand why there are two different MDI icon fonts in this package
> 
> That's because mdi is for version 5.x and mdi6 is for version 6.x (similar to the management we have for FontAwesome: fa is version 4.x and fa5* is for version 5.x).

Ah, I see! I've had a look at the MDI website, and they do, indeed, have breaking changes between major versions. But there are also licensing issues that they are addressing between versions, for example: https://github.com/Templarian/MaterialDesign/issues/5797 about removing the Adobe icons. They're also planning on removing all brand icons in version 8 some time in late 2023.

Stability means that you want to continue providing old versions of the fonts, though, so that code which uses qtawesome and refers to an icon in MDI doesn't break when MDI changes the icon names.  But it does mean that qtawesome will keep growing every time you include a new MDI major or minor version.  (They say that they try to avoid breaking changes between minor versions, though, so it's probably OK to replace the MDI v6 font in qtawesome with version 6.9.96.)

**Comment by dalthviz** [comment]
> That's an interesting idea but I don't know if it'd work without being online. So, I'm not sure about it.

Not totally sure to get the _work without being online_ part either but just in case the idea would be something like if you want to use the FontAwesome 5 you will need to run something like `pip install qtawesome qtawesome-fa5` instead of `pip install qtawesome` (maybe even just something like `pip install qtawesome-fa5` since most probably we could set for `qtawesome-fa5` or any other font package a dependency to `qtawesome`)

>  But it does mean that qtawesome will keep growing every time you include a new MDI major or minor version.

Yep, you are totally right there, one of the alternatives we though for that is to just bundle at most to major versions per font (so for example if we want to bundle FontAwesome 6.x, we will need to remove first FontAwesome 4.x )

> They say that they try to avoid breaking changes between minor versions, though, so it's probably OK to replace the MDI v6 font in qtawesome with version 6.9.96

I will create an issue to track that missing update for `mdi6` :+1:
